gateway/handle.go

```diff
@@ -670,6 +671,16 @@ func (gw *Handle) internalBatchHandlerFunc() http.HandlerFunc {
 				stats.CountType,
 				gw.newReqTypeStatsTagsWithReason(reqType, ""),
 			).Count(len(jobs))
+
+			// Sending events to config backend
+			for _, job := range jobs {
+				writeKey, ok := jobsUUIDToWriteKeyMap[job.UUID]
```

Comment:
What if we call `gw.writeKeyForSourceID` here instead of passing the `jobsUUIDToWriteKeyMap` ?

Reply:
do you mean to say, we get the source ID from params and call `writeKeyForSourceID` with sourceID?